### PR TITLE
Call refreshViews after every restart

### PR DESCRIPTION
**Pull request by johnmccutchan**
- [x] Until I sort out asynchronous native embedder service protocol requests, always refresh the view list after a restart.
- [x] Add help text about SIGUSR1/SIGUSR2

### CODE REVIEW

**Comment by johnmccutchan** [comment]
@Hixie

File: packages/flutter_tools/lib/src/commands/run.dart [hunk 1]
```diff
@@ -70,7 +70,9 @@ class RunCommand extends RunCommandBase {
 
     // Option to write the pid to a file.
     argParser.addOption('pid-file',
-                        help: 'Specify a file to write the process id to.');
+                        help: 'Specify a file to write the process id to.'
+                              'NOTE: You can send SIGUSR1 to trigger a hot '
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/run.dart]: Remove the word "NOTE"

File: packages/flutter_tools/lib/src/commands/run.dart [hunk 2]
```diff
@@ -70,7 +70,9 @@ class RunCommand extends RunCommandBase {
...
+                              'reload and SIGUSR2 to trigger a full restart');
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/run.dart]: period at the end

File: packages/flutter_tools/lib/src/resident_runner.dart [hunk 3]
```diff
@@ -47,10 +47,16 @@ abstract class ResidentRunner {
   }
 
   Future<Null> _debugDumpApp() async {
+    if (vmService != null) {
+      await vmService.vm.refreshViews();
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/resident_runner.dart]: no braces

File: packages/flutter_tools/lib/src/resident_runner.dart [hunk 4]
```diff
@@ -47,10 +47,16 @@ abstract class ResidentRunner {
...
+    }
     await currentView.uiIsolate.flutterDebugDumpApp();
   }
 
   Future<Null> _debugDumpRenderTree() async {
+    if (vmService != null) {
+      await vmService.vm.refreshViews();
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/resident_runner.dart]: ditto

**Comment by Hixie** [comment]
LGTM

File: packages/flutter_tools/lib/src/commands/run.dart (hunk 1 above)
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/commands/run.dart]: Done.

File: packages/flutter_tools/lib/src/commands/run.dart (hunk 2 above)
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/commands/run.dart]: Done.

File: packages/flutter_tools/lib/src/resident_runner.dart (hunk 3 above)
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/resident_runner.dart]: Done.

File: packages/flutter_tools/lib/src/resident_runner.dart (hunk 4 above)
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/resident_runner.dart]: Done.

**Pull request merged**